graph-manager-docs/source/federation.mdx

```diff
@@ -123,9 +123,115 @@ server.listen().then(({ url }) => {
+
+1. Register a new `email` service that removes the `EmailChannel` implementation of `Channel`
+1. Deploy a new Email service that removes the implementation
+
+The first step will cause the query-planner to stop sending along fragments `...on EmailChannel`, which would fail validation if sent
+to a service that did not have awareness of that type!
+
+#### Modifying query-planning logic
+
+You'll want to treat migrations of query planning logic similarly to how you may treat database migrations, by carefully considering the effects on the
+downstream services as the query planner changes, and planning for "double reading" as appropriate. For example, consider the case of migrating a
+`@required` field in an extension from one to another. In this example, we'll consider the `Products` service and the `Reviews` service, like so:
+
+```js
+// Products Service
+
+type Product @key(fields: "upc") {
+  upc: ID!
+  nameLowerCase: String!
+}
+
+// Review Service
+
+extend type Product @key(fields: "upc") {
+  upc: ID! @external
+  reviews: [Review]! @requires(fields: "nameLowercase")
+  nameLowercase: String! @external
+}
+```
+
+If the `nameLowercase` field is to be deprecated and replaced with the `name` field
+
+```js
+// Products Service
+
+type Product @key(fields: "upc") {
+  upc: ID!
+  nameLowerCase: String! @deprecated
+  name: String!
+}
+
+// Reviews Service
+
+extend type Product @key(fields: "upc") {
+  upc: ID! @external
+  nameLowercase: String! @external
+  name: String! @external
+  reviews: [Review]! @requires(fields: "name")
+}
+```
+
+To do this migration in-place, you would perform the following steps:
+
+1. Modify the `Products` service to add the new field (as per usual, first deploy all replicas, then `apollo service:push` the new partial schema)
+1. Deploy a new version of the `Reviews` service with a resolver that accepts _either_ `nameLowercase` or `name` in the source object
+1. Modify the schema of the `Reviews` service in the registry to update query-planning logic to `@require` the new `name` field
+1. Deploy a new version of the `Reviews` service with a resolver that only accepts the `name` in its source object
+
+Alternatively, you could perform this operation with an atomic migration at the service level, by modifying the URL of the service. For that, the
+steps would be:
+
+1. Modify the `Products` service to add the `name` field (as per usual, first deploy all replicas, then `apollo service:push` the new partial schema)
+1. Deploy a new set of `Reviews` replicas to a new URL
+1. Register the `Reviews` service with the new URL and the partial schema changes above
+
+With this atomic strategy, the query planner will resolve all outstanding requests to the old service that relied on `nameLowercase` with the old
+query-planning configuration, which `@required` the `nameLowercase` field, and all **new** requests would be made to the new service (due to the
```

Comment:
Again, avoid alternate forms of directives like `required`

"would be made" -> "are made"